autoPyTorch/pipeline/components/training/trainer/__init__.py

```diff
@@ -351,6 +352,16 @@ def _fit(self, X: Dict[str, Any], y: Any = None, **kwargs: Any) -> 'TrainerChoic
             )
             self.save_model_for_ensemble()
 
+        # As training have finished, load the best weight
+        if self.checkpoint_dir is not None:
+            best_path = os.path.join(self.checkpoint_dir, 'best.pth')
+            self.logger.debug(f" Early stopped model {X['num_run']} on epoch {self.run_summary.get_best_epoch()}")
+            # We will stop the training. Load the last best performing weights
+            X['network'].load_state_dict(torch.load(best_path))
+
+            # Clean the temp dir
+            shutil.rmtree(self.checkpoint_dir)
```

Comment:
Probably we need to refactor this file in the future.
But just for the start, could you make this method?
Just in case, Dict is passed by reference, so it will not be a problem. 
```
        # As training have finished, load the best weight
        def _load_best_weights_and_clean_checkpoints(self, X: Dict[str, Any]) -> None:
            """
            Load the best model until the last epoch and delete all the files for checkpoints.

            Args:
                X (Dict[str, Any]): Dependencies needed by current component to perform fit
           """
            best_path = os.path.join(self.checkpoint_dir, 'best.pth')
            self.logger.debug(f" Early stopped model {X['num_run']} on epoch {self.run_summary.get_best_epoch()}")
            # We will stop the training. Load the last best performing weights
            X['network'].load_state_dict(torch.load(best_path))

            # Clean the temp dir
            shutil.rmtree(self.checkpoint_dir)
```